src/pygeom.c

```diff
@@ -97,6 +97,61 @@ static PyObject *GeometryObject_ToWKT(GeometryObject *obj, char *format)
         }
 }
 
+
+static PyObject *GeometryObject_ToWKB(GeometryObject *obj)
+{
+    unsigned char *wkb = NULL;
+    char has_empty = 0;
+    size_t size;
+    PyObject *result = NULL;
+    GEOSGeometry *geom = NULL;
+    GEOSWKBWriter *writer = NULL;
+    if (obj->ptr == NULL) {
+        Py_INCREF(Py_None);
+        return Py_None;
+    }
+
+    GEOS_INIT;
+
+    // WKB Does not allow empty points.
+    // We check for that and patch the POINT EMPTY if necessary
+    has_empty = has_point_empty(ctx, obj->ptr);
+    if (has_empty == 2) { errstate = PGERR_GEOS_EXCEPTION; goto finish; }
+    if (has_empty) {
+        geom = point_empty_to_nan_all_geoms(ctx, obj->ptr);
+    } else {
+        geom = obj->ptr;
+    }
+
+    /* Create the WKB writer */
+    writer = GEOSWKBWriter_create_r(ctx);
+    if (writer == NULL) { errstate = PGERR_GEOS_EXCEPTION; goto finish; }
+    // Allow 3D output and include SRID
+    GEOSWKBWriter_setOutputDimension_r(ctx, writer, 3);
```

Comment:
It doesn't look like the 3rd dimension is tested; geometries with Z value are not included in the `all_types` used for test inputs to pickle tests.